src/error.ts

```diff
@@ -1,12 +1,22 @@
 /** @public */
 export class BSONError extends Error {
+  constructor(m: string) {
+    super(m);
```

Comment:
```suggestion
  constructor(message: string) {
    super(message);
```

IMO, Just preferable to be explicit. (Same for the other class)